coral-spark/src/test/java/com/linkedin/coral/spark/CoralSparkTest.java

```diff
@@ -382,11 +383,16 @@ public void testSchemaPromotionView() {
   @Test
   public void testUnionExtractUDF() {
     RelNode relNode = TestUtils.toRelNode("SELECT extract_union(foo) from union_table");
-    String targetSql = String.join("\n", "SELECT foo", "FROM default.union_table");
+    String targetSql = String.join("\n", "SELECT coalesce_struct(foo)", "FROM default.union_table");
     assertEquals(CoralSpark.create(relNode).getSparkSql(), targetSql);
 
-    RelNode relNode2 = TestUtils.toRelNode("SELECT extract_union(foo, 2) from union_table");
-    String targetSql2 = String.join("\n", "SELECT foo.tag_2", "FROM default.union_table");
+    RelNode relNode1 = TestUtils.toRelNode("SELECT extract_union(foo, 2) from union_table");
+    String targetSql1 = String.join("\n", "SELECT coalesce_struct(foo, 3)", "FROM default.union_table");
+    assertEquals(CoralSpark.create(relNode1).getSparkSql(), targetSql1);
+
+    // Recursion case
+    RelNode relNode2 = TestUtils.toRelNode("SELECT extract_union(a) from recursive_union");
+    String targetSql2 = String.join("\n", "SELECT coalesce_struct(a)", "FROM default.recursive_union");
     assertEquals(CoralSpark.create(relNode2).getSparkSql(), targetSql2);
```

Comment:
I don't think the recursion case here is necessary given it does not verify the type of translated SQL's relNode, previous 2 test cases should be fine.
I think we need to test the type of RelNode from SQL like `SELECT coalesce_struct(a) FROM default.recursive_union` where `a` is nested union as mentioned above.